src/main/java/com/example/basicbookstoreprojectnew/model/service/bookserviceimpl/BookServiceImpl.java

```diff
@@ -0,0 +1,60 @@
+package com.example.basicbookstoreprojectnew.model.service.bookserviceimpl;
+
+import com.example.basicbookstoreprojectnew.dto.BookDto;
+import com.example.basicbookstoreprojectnew.dto.CreateBookRequestDto;
+import com.example.basicbookstoreprojectnew.mapper.BookMapper;
+import com.example.basicbookstoreprojectnew.model.Book;
+import com.example.basicbookstoreprojectnew.model.repository.BookRepository;
+import com.example.basicbookstoreprojectnew.model.service.BookService;
+import jakarta.persistence.EntityNotFoundException;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@Transactional
+@RequiredArgsConstructor
+public class BookServiceImpl implements BookService {
+    private final BookRepository bookRepository;
+    private final BookMapper bookMapper;
+
+    @Override
+    public BookDto save(CreateBookRequestDto createBookRequestDto) {
+        Book book = bookMapper.toEntity(createBookRequestDto);
+        bookRepository.save(book);
+        return bookMapper.toDto(book);
+    }
+
+    @Override
+    @Transactional(readOnly = true)
+    public List<BookDto> findAll() {
+        return bookRepository.findAll().stream()
+                .map(bookMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public BookDto findById(Long id) {
+        Book book = bookRepository.findById(id)
+                .orElseThrow(() -> new RuntimeException("Book not found! "));
```

Comment:
provide parameter to message
```suggestion
                .orElseThrow(() -> new EntityNotFoundException("Book not found! "));
```